source/core/shape.cc

```diff
@@ -635,17 +639,20 @@ RBFShape<dim>::RBFShape(const std::vector<double> &shape_arguments,
                orientation)
 // The effective radius is extracted at the proper index from shape_arguments
 {
-  number_of_nodes = shape_arguments.size() / (dim + 3);
+  number_of_nodes = (shape_arguments.size() - 1) / (dim + 3);
```

Comment:
Can you comment on this calculation? it seems a bit random? Explain the origin of this line.